osu.Game/Screens/Play/SubmittingPlayer.cs

```diff
@@ -106,27 +108,16 @@ protected override async Task PrepareScoreForResultsAsync(Score score)
         {
             await base.PrepareScoreForResultsAsync(score).ConfigureAwait(false);
 
-            // token may be null if the request failed but gameplay was still allowed (see HandleTokenRetrievalFailure).
-            if (token == null)
-                return;
-
-            var tcs = new TaskCompletionSource<bool>();
-            var request = CreateSubmissionRequest(score, token.Value);
+            await submitScore(score).ConfigureAwait(false);
+        }
 
-            request.Success += s =>
-            {
-                score.ScoreInfo.OnlineScoreID = s.ID;
-                tcs.SetResult(true);
-            };
+        public override bool OnExiting(IScreen next)
+        {
+            var exiting = base.OnExiting(next);
 
-            request.Failure += e =>
-            {
-                Logger.Error(e, "Failed to submit score");
-                tcs.SetResult(false);
-            };
+            submitScore(Score);
```

Comment:
This is slightly scary, albeit I do understand not wanting to block... I'm not necessarily seeing a way in which this could fail, although one thing that may end up being a bit dodgy is the `CreateSubmissionRequest` call in `submitScore`. The hope would be that that method reads the current beatmap before player loader has returned its lease / any other component can plausibly change the beatmap to another one.

Reply:
The hope is that the score is copied at the point of the method call, yep.

That said, i'm not seeing how a beatmap change would possible affect the `Score`?

Reply:
I was more worried about the fact that `SubmitSoloScoreRequest` is sourcing its `beatmapId` from the game-global bindable:

https://github.com/ppy/osu/blob/5a23c4b10612056ba74762ee4c03bd0ce1009633/osu.Game/Screens/Play/SoloPlayer.cs#L28-L35

Really should be using score if plausible there, now that I think about it:

Reply:
This is all run synchronously regardless, so I don't believe it is an issue, but I'll use the local `beatmap` regardless.